lib/bsdlib/nrf91_sockets.c

```diff
@@ -869,15 +883,19 @@ static ssize_t nrf91_socket_offload_sendmsg(void *obj, const struct msghdr *msg,
 			continue;
 		}
 
-		ret = nrf91_socket_offload_sendto(obj, msg->msg_iov[i].iov_base,
-						  msg->msg_iov[i].iov_len,
-						  flags, msg->msg_name,
-						  msg->msg_namelen);
-		if (ret < 0) {
-			return ret;
+		bytes_sent_this_buffer = 0;
+		while (bytes_sent_this_buffer < msg->msg_iov[i].iov_len) {
+			ret = nrf91_socket_offload_sendto(obj,
+							(msg->msg_iov[i].iov_base + bytes_sent_this_buffer),
```

Comment:
What tab size and line length limits should I be using?  My editor tries to be smart and auto-guess each time I open a file, but I feel like it messed up here since I just realized it's giving me a tab size of 3 in this file.

Reply:
8 spaces per tab, 80 chars per line. Crazy, I know.